### PR TITLE
recieve response without request being another argument in handler

### PR DESCRIPTION
**Pull request by ayaankhan98**
Linked Issue #47

### CODE REVIEW

File: include/crow/http_response.h [hunk 1]
```diff
@@ -124,6 +124,11 @@ namespace crow
             }
         }
 
+        void end(const int& _code) {
+            code = _code;
+            end();
+        }
+
```
Comment (The-EDev) [inline include/crow/http_response.h:131]: This is unrelated to the PR, and as far as I know not exactly a problem. It's probably best not to clutter the response class as much.

**Review by The-EDev — CHANGES_REQUESTED** [review]

File: include/crow/http_response.h (hunk 1 above)
Comment (ayaankhan98) [reply include/crow/http_response.h:131]: No problem @The-EDev  i will remove this. I am still not familiar with the complete codebase that's why i added this too from the old repository PR.

Comment (The-EDev) [reply include/crow/http_response.h:131]: Also it might be best to change the examples to reflect this change?

Comment (The-EDev) [reply include/crow/http_response.h:131]: Also, trying to build the examples without a request parameter resulted in `error: call of ‘(crow::TaggedRule<long int, long int>) (main()::<lambda(crow::response&, int, int)>)’ is ambiguous`

Comment (ayaankhan98) [reply include/crow/http_response.h]: okay i will see this and will also add an example to reflect these changes

Comment (The-EDev) [reply include/crow/http_response.h]: You probably don't need to add an example, rather change the existing ones that have a request that's not being used

Comment (ayaankhan98) [reply include/crow/http_response.h]: yes, will modify the existing one

File: examples/example.cpp [hunk 2]
```diff
@@ -13,12 +13,11 @@ struct ExampleMiddleware
 {
     std::string message;
 
-    ExampleMiddleware() 
+    ExampleMiddleware() : message(std::string("foo"))
```
Comment (mrozigor) [inline examples/example.cpp:16]: `message("foo")`?

Comment (The-EDev) [reply examples/example.cpp:16]: originally this was defined the the body (2 lines below)

Comment (ayaankhan98) [reply examples/example.cpp:16]: earlier i tried codacy analysis with crow there was just 2-3 minor issues, instead of creating a separate PR for such minor things i thought to pick them along with this, that's why i changed this

Comment (mrozigor) [reply examples/example.cpp:16]: Yeah, I know. I had on mind change from `message(std::string("foo"))` to `message(."foo")`

File: include/crow/routing.h [hunk 3]
```diff
@@ -568,6 +594,7 @@ namespace crow
 
     private:
         std::function<void(const crow::request&, crow::response&, Args...)> handler_;
+        std::function<void(crow::response&, Args...)> handler_without_request;
```
Comment (ayaankhan98) [inline include/crow/routing.h:597]: added this just to prevent that `unsed variable warning in lambda`

**Review by The-EDev — COMMENTED** [review]
Getting this error when running example
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  no handler for url /add/<int>/<int>
Aborted (core dumped)
```

**Comment by ayaankhan98** [comment]
> Getting this error when running example
> 
> ```
> terminate called after throwing an instance of 'std::runtime_error'
>   what():  no handler for url /add/<int>/<int>
> Aborted (core dumped)
> ```

sorry my mistake this was due to
```
void validate() override
        {
            if (!handler_)
            {
                throw std::runtime_error(name_ + (!name_.empty() ? ": " : "") + "no handler for url " + rule_);
            }
        }
```
i tried to fix this by adding 
```
void validate() override
        {
            if (!handler_ && !handler_without_request)
            {
                throw std::runtime_error(name_ + (!name_.empty() ? ": " : "") + "no handler for url " + rule_);
            }
        }
```
but after this on `add/int/int` route there was `500 internal server error`, therefore i removed that extra varieble `handler_without_request`. Things are working now but with at warning of `unsed variable req in lambda`

**Comment by ayaankhan98** [comment]
> Getting this error when running example
> 
> ```
> terminate called after throwing an instance of 'std::runtime_error'
>   what():  no handler for url /add/<int>/<int>
> Aborted (core dumped)
> ```

@The-EDev fixed

File: examples/example.cpp (hunk 2 above)
Comment (The-EDev) [reply examples/example.cpp:16]: the example seems to work fine without the std::string section, seems unnecessary, could you remove it?

Comment (The-EDev) [reply examples/example.cpp:16]: Also line 86 still contains a request parameter, could you remove it?

File: examples/example_vs.cpp [hunk 4]
```diff
@@ -13,12 +13,11 @@ struct ExampleMiddleware
 {
     std::string message;
 
-    ExampleMiddleware() 
+    ExampleMiddleware() : message(std::string("foo"))
```
Comment (The-EDev) [inline examples/example_vs.cpp:16]: again could you remove the std::string portion

**Review by The-EDev — CHANGES_REQUESTED** [review]
in addition to the other comments, I would be thankful if you could edit line 33 and 34 of `docs/guides/routes.md` (remove line 34, edit the code part of 33, and add a `<br>` at the end of 33).

Everything else seems good, thanks for your help.

File: examples/example_vs.cpp (hunk 4 above)
Comment (ayaankhan98) [reply examples/example_vs.cpp:16]: yes removed message was already declared as std::string so there is no need to typecaste to std::string again

File: examples/example.cpp (hunk 2 above)
Comment (ayaankhan98) [reply examples/example.cpp]: fixed both issues

**Comment by ayaankhan98** [comment]
> in addition to the other comments, I would be thankful if you could edit line 33 and 34 of `docs/guides/routes.md` (remove line 34, edit the code part of 33, and add a `<br>` at the end of 33).
> 
> Everything else seems good, thanks for your help.

Done!

File: docs/guides/routes.md [hunk 5]
```diff
@@ -30,8 +30,7 @@ You can also access the url parameters in the handler using `#!cpp req.url_param
 For more information on `crow::request` go [here](../../reference/structcrow_1_1request.html).<br><br>
 
 ###Response
-Crow also provides the ability to define a response in the parameters by using `#!cpp ([](const crow::request& req, crow::response& res){...})`.<br>
-If you don't want to use the request you can write `#!cpp ([](const crow::request& , crow::response& res){...})`.<br><br>
+Crow also provides the ability to define a response in the parameters by using `#!cpp ([](crow::response& res){...})`.<br>
```
Comment (The-EDev) [inline docs/guides/routes.md:33]: there's a second `<br>` missing here

Comment (ayaankhan98) [reply docs/guides/routes.md]: added

**Review by The-EDev — APPROVED** [review]
